guardian/prosemirror-elements
# Add more control over repeater child nodes

Paired on with @twrichards 

## What does this change?
Previously we were able to add child nodes at the end of the repeater, and remove child nodes at a given index.

This PR allows us to add child nodes at a given index, and move child nodes up or down an index. It also refactors the remove method to have greater guards against invalid invocation (in line with the new methods), and renames the methods to be more precise.

## How to test
You can play about with these controls by creating an Alt Style element in the demo:
![image](https://github.com/guardian/prosemirror-elements/assets/40991816/7975fdca-649e-4455-bedf-84791dca6068)

Check that the controls work as expected, and are enabled / disabled in the right places.

There are also new tests for these methods which can be inspected and run to give confidence in the PR's intent.
